src/uu/ps/src/mapping.rs

```diff
@@ -0,0 +1,190 @@
+// This file is part of the uutils procps package.
+//
+// For the full copyright and license information, please view the LICENSE
+// file that was distributed with this source code.
+
+use crate::parser::OptionalKeyValue;
+use std::collections::HashMap;
+
+pub(crate) fn apply_format_mapping(formats: &[OptionalKeyValue]) -> HashMap<String, String> {
+    let mut mapping = default_mapping();
+
+    for optional_kv in formats {
+        let key = optional_kv.key();
+        if !optional_kv.is_value_empty() {
+            mapping.insert(key.to_owned(), optional_kv.try_get::<String>().unwrap());
+        }
+    }
+
+    mapping
+}
+
+/// This function will extract all the needed headers from matches (the data being needed)
+///
+/// The headers are sequential, and the order about the final output is related to the headers
+pub(crate) fn default_codes() -> Vec<String> {
+    let mut mapping = Vec::new();
+    let mut append = |code: &str| mapping.push(code.into());
+
+    append("pid");
+    append("tname");
+    append("time");
+    append("ucmd");
+
+    mapping
```

Comment:
While this code works fine, it is a bit difficult to read. My suggestion is to use something like
```
vec!["pid".into(), "tname".into(), ...]`
```
or 
```
["pid", "tname", "time", "ucmd"].map(Into::into).to_vec()
```